src/store/useTracksStore.ts

```diff
@@ -38,190 +38,195 @@ const useTracksStore = create<TracksState>((set, get) => ({
     const { currentPage, itemsPerPage, sort, filter } = get();
     set({ isLoading: true });
 
-    try {
-      const response = await fetchTracks(
-        currentPage,
-        itemsPerPage,
-        sort.field as string,
-        sort.direction,
-        filter.search,
-        filter.genres.length ? filter.genres[0] : undefined,
-      );
+    const result = await fetchTracks(
+      currentPage,
+      itemsPerPage,
+      sort.field,
+      sort.direction,
+      filter.search,
+      filter.genres.length ? filter.genres[0] : undefined
+    );
 
+    if (result.isOk()) {
       set({
-        tracks: response.data,
-        totalTracks: response.meta.total,
-        totalPages: response.meta.totalPages,
+        tracks: result.value.data,
+        totalTracks: result.value.meta.total,
+        totalPages: result.value.meta.totalPages,
         isLoading: false,
       });
-    } catch (error) {
+    } else {
       set({ isLoading: false });
-      console.error("Failed to fetch tracks:", error);
+      console.error("Failed to fetch tracks:", result.error);
     }
   },
 
   fetchAllGenres: async () => {
-    try {
-      const genres = await fetchGenres();
-      set({ genres });
-    } catch (error) {
-      console.error("Failed to fetch genres:", error);
+    const result = await fetchGenres();
+    if (result.isOk()) {
+      set({ genres: result.value });
+    } else {
+      console.error("Failed to fetch genres:", result.error);
+      set({ genres: [] });
     }
```

Comment:
You can add `return`  in if block, to reduce nesting
```suggestion
    if (result.isOk()) {
      set({ genres: result.value });
      return
    } 
    
      console.error("Failed to fetch genres:", result.error);
      set({ genres: [] });
    
```